### PR TITLE
Update README.md to include comprehensive documentation for the NEAT-…

### PR DESCRIPTION
**Pull request by nleck**
…AI-Discovery Rust library, detailing its goals, problem statement, performance requirements, features, development guidelines, file format, JSON interface, code quality checks, cross-platform support, and versioning. This update aims to clarify the library's purpose and usage for developers.

---

> [!NOTE]
> Introduce Rust crate to record discovery data to Parquet with JSON interface, tests, and a full CI/CD + security workflow.
> 
> - **Library (Rust)**:
>   - Implement discovery recording crate (`src/lib.rs`, `src/record.rs`, `src/parquet_format.rs`, `src/types.rs`) with JSON I/O and Parquet writing via Arrow/Parquet.
>   - Add unit and integration tests (`tests/`), including validation for indices and overflow safety.
> - **CI/CD & Security**:
>   - Add multi-job workflow `.github/workflows/ci.yml` (auto-format, version bump on `src/` changes, quality checks, build, shell checks, spell check).
>   - Add reusable security workflow `.github/workflows/security.yml` (cargo-audit, dependency review).
> - **Tooling & Scripts**:
>   - Add `quality.sh` for local checks and `scripts/runlib.sh` for versioned library builds/install.
> - **Project Setup**:
>   - Create `Cargo.toml` with dependencies (`serde`, `anyhow`, `arrow`, `parquet`, etc.).
>   - Expand `README.md` with goals, format specs, JSON interface, and workflows.
>   - Update `.gitignore` (ignore `Cargo.lock`, build artifacts).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b1b4912696922e99af2bb548a63b7c08f0e70a9c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+        fi
+
+  build:
+    name: Build Release
+    runs-on: ubuntu-latest
+    needs: [version-increment, auto-format, quality]
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
```
[the remark this answers is not in the document]
Comment (stservice) [reply .github/workflows/ci.yml:274]: fixed
